From 3c29733b9670dd6c99ff58df79f6d32f4ceb9e6a Mon Sep 17 00:00:00 2001
From: Conor McCarthy <cmccarthy@codeweavers.com>
Date: Wed, 1 Jul 2026 10:50:04 +1000
Subject: [PATCH 31/48] windows.storage: Implement InMemoryRandomAccessStream
 IRandomAccessStream::get_Position().

(cherry picked from commit eb4a2138c1e3a9b39d53fcc1b7bf2cd1d0da6389)

CW-Bug-Id: #27221
---
 dlls/windows.storage/streams.c       | 5 +++--
 dlls/windows.storage/tests/storage.c | 6 ------
 2 files changed, 3 insertions(+), 8 deletions(-)

diff --git a/dlls/windows.storage/streams.c b/dlls/windows.storage/streams.c
index d61788fc0a8..70e8de96684 100644
--- a/dlls/windows.storage/streams.c
+++ b/dlls/windows.storage/streams.c
@@ -343,9 +343,10 @@ static HRESULT WINAPI memory_stream_random_access_get_Position( IRandomAccessStr
 {
     struct memory_stream *impl = impl_from_IRandomAccessStream( iface );
 
-    FIXME( "iface %p, value %p stub!\n", iface, value );
+    TRACE( "iface %p, value %p.\n", iface, value );
 
-    return impl->closed ? RO_E_CLOSED : E_NOTIMPL;
+    *value = impl->pos;
+    return impl->closed ? RO_E_CLOSED : S_OK;
 }
 
 static HRESULT WINAPI memory_stream_random_access_Seek( IRandomAccessStream *iface, UINT64 position )
diff --git a/dlls/windows.storage/tests/storage.c b/dlls/windows.storage/tests/storage.c
index b9648582f50..527b85ee627 100644
--- a/dlls/windows.storage/tests/storage.c
+++ b/dlls/windows.storage/tests/storage.c
@@ -360,7 +360,6 @@ static void test_InMemoryRandomAccessStream(void)
     ok( hr == S_OK, "got hr %#lx.\n", hr );
     ok( value64 == 0, "got size %I64u.\n", value64 );
     hr = IRandomAccessStream_get_Position( in_memory_stream, &value64 );
-    todo_wine
     ok( hr == S_OK, "got hr %#lx.\n", hr );
     ok( value64 == 0, "got pos %I64u.\n", value64 );
     hr = IRandomAccessStream_Seek( in_memory_stream, 16 );
@@ -370,7 +369,6 @@ static void test_InMemoryRandomAccessStream(void)
     ok( hr == S_OK, "got hr %#lx.\n", hr );
     ok( value64 == 0, "got size %I64u.\n", value64 );
     hr = IRandomAccessStream_get_Position( in_memory_stream, &value64 );
-    todo_wine
     ok( hr == S_OK, "got hr %#lx.\n", hr );
     todo_wine
     ok( value64 == 16, "got pos %I64u.\n", value64 );
@@ -392,7 +390,6 @@ static void test_InMemoryRandomAccessStream(void)
     todo_wine
     ok( hr == S_OK, "got hr %#lx.\n", hr );
     hr = IRandomAccessStream_get_Position( in_memory_stream, &value64 );
-    todo_wine
     ok( hr == S_OK, "got hr %#lx.\n", hr );
     todo_wine
     ok( value64 == 0, "got pos %I64u.\n", value64 );
@@ -403,7 +400,6 @@ static void test_InMemoryRandomAccessStream(void)
     todo_wine
     ok( value64 == 17, "got size %I64u.\n", value64 );
     hr = IRandomAccessStream_get_Position( in_memory_stream, &value64 );
-    todo_wine
     ok( hr == S_OK, "got hr %#lx.\n", hr );
     todo_wine
     ok( value64 == 8, "got pos %I64u.\n", value64 );
@@ -444,7 +440,6 @@ static void test_InMemoryRandomAccessStream(void)
     todo_wine
     ok( value == 17, "got read %u.\n", value );
     hr = IRandomAccessStream_get_Position( in_memory_stream, &value64 );
-    todo_wine
     ok( hr == S_OK, "got hr %#lx.\n", hr );
     todo_wine
     ok( value64 == 17, "got pos %I64u.\n", value64 );
@@ -496,7 +491,6 @@ static void test_InMemoryRandomAccessStream(void)
     ok( hr == S_OK, "got hr %#lx.\n", hr );
     ok( value64 == 0x100000, "got size %I64u.\n", value64 );
     hr = IRandomAccessStream_get_Position( in_memory_stream, &value64 );
-    todo_wine
     ok( hr == S_OK, "got hr %#lx.\n", hr );
     todo_wine
     ok( value64 == 18, "got pos %I64u.\n", value64 );
-- 
2.55.0

